specifications/specifications.tex

```diff
@@ -71,47 +71,73 @@ \subsection{Ownership Transfer}
+    \end{itemize}
+    \item the multicall is approved by the owner and a majority of guardians see Section \ref{sec:guardian-approval} 
+    \item the multicall is executed with a valid session key see Section \ref{sec:session}
+\end{enumerate}
 
-The daily spend of a user is reset after 24 hours or after each transaction made via the \emph{ApprovedTransfer} feature, which involves a majority of guardians (see Section~\ref{sec:approved-transfer}).
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{Trusted Contacts}
+\label{sec:trusted-contacts}
+
+The wallet keeps a list of trusted addresses managed by the wallet owner.
+Adding an address to this list is an action triggered by the wallet owner and takes 36 hours to be effective. Removing an address is triggered by the owner and is immediate.
 
-The daily limit is cross-token (ETH + ERC20) and we're using an on-chain oracle to get the conversion rates for ERC20 tokens.
+Interracting with a trusted contact to e.g. send an asset, can be triggered by the wallet owner and does not require any additional authorisation. 
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-\subsection{Whitelist}
-\label{sec:whitelist}
+\subsection{Dapp Registry}
+\label{sec:authorised-dapps}
+
+In addition to trusted contacts, the wallet supports 2 registries of addresses that have been authorised by Argent.
 
-The wallet keeps a whitelist of trusted addresses. Transfers to those addresses are immediate and their amounts are not limited.
+The first registry is enabled by default (opt-out) for all wallets and contains all the dapps that are natively integrated in the Argent client application (Paraswap, Compound, Maker, Aave, etc). The second registry is disabled by default (opt-in) and contains a list of curated dapps that the user can access through WalletConnect.
 
-Adding an address to the whitelist is an action triggered by the wallet owner and takes 24 hours to be effective. Removing an address is triggered by the owner and is immediate.
+Each registry entry is a mapping between an authorised address and a filter containing the set of conditions that a transaction from a wallet to the authorised address must satisfy.
+Adding or removing an entry in a registry is a timelocked operations that can only be initiated by the Argent multisig.
+The timelock is set to 7 days to give enough time for a user to disable the correspoding registry should they not approve the new addition. 
+
+Enabling a registry for a wallet requires guardian approvals and the number of required signatures is given by
+\begin{equation*}
+    1+\left\lceil {\frac{n}{2}} \right\rceil
+\end{equation*}
+where the first signature is the owner and $n$ is the total number of guardians. Disabling a registry can be executed by the wallet owner alone.
```

Comment:
Makes sense but just a reminder that for this to work like that we will need to authorise the `DappRegistry` itself with a `DappRegistryFilter` that only allows calls to `toggleRegistry` with `_enabled=false`.

Reply:
Good point, and let's not complexity the architecture. I'll update the spec so that disabling a registry is a guardian approved operation as well.